source/simulator/assembly.cc

```diff
@@ -1521,7 +1521,7 @@ namespace aspect
       }
     else
       {
-        compute_material_model_input_values (old_solution,
+        compute_material_model_input_values (current_linearization_point,
```

Comment:
Yes, in principle we could have the lines outside of the if() else() statement now. However, I think this change is only one step towards removing the whole if-loop. The explicit material model inputs are used in the stabilization to compute the advection system residual, and right now this computation only includes the temperature terms. I think at some point we also want to use the reaction term there, and then we will also need the explicit material model inputs for the composition. 
But this might take some time to test, and so I haven't included it in this pull request yet....